### ISSUE TITLE
[terra-clinical-item-collection][terra-clinical-item-display] Accessibilities errors reported by axe-core

### ISSUE THREAD

**Issue by benbcai** [cerner/terra-clinical#643]
# Bug Report
terra-clinical-item-collection and terra-clinical-item-display have accessibility failures in wdio tests when running agains axe-core 3.5.0

## Description
- terra-clinical-item-collection - A selectable List with the `listbox` role requires an `aria-label`

- terra-clinical-item-display - A disabled item display has an `opacity` style of `0.2` and this causes insufficient color-contrast. `aria-disabled` needs to be set for a disabled item display.

## Steps to Reproduce
1. Pull in axe-core 3.5.0 and run the wdio tests to see the failures

## Additional Context / Screenshots

> expected no accessibility violations but got:
> 	{
>   "description": "Ensures every ARIA input field has an accessible name",
>   "help": "ARIA input fields have an accessible name",
>   "helpUrl": "https://dequeuniversity.com/rules/axe/3.5/aria-input-field-name?application=axeAPI",
>   "id": "aria-input-field-name",
>   "impact": "serious",
>   "nodes": [
>     {
>       "all": [],
>       "any": [
>         {
>           "data": null,
>           "id": "aria-label",
>           "impact": "serious",
>           "message": "aria-label attribute does not exist or is empty",
>           "relatedNodes": []
>         },
>         {
>           "data": null,
>           "id": "aria-labelledby",
>           "impact": "serious",
>           "message": "aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty",
>           "relatedNodes": []
>         },
>         {
>           "data": null,
>           "id": "non-empty-title",
>           "impact": "serious",
>           "message": "Element has no title attribute or the title attribute is empty",
>           "relatedNodes": []
>         }
>       ],
>       "failureSummary": "Fix any of the following:\n  aria-label attribute does not exist or is empty\n  aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty\n  Element has no title attribute or the title attribute is empty",
>       "html": "<ul data-terra-clinical-item-collection-list-view=\"true\" role=\"listbox\" class=\"List-module__list___4A7Nh\">",
>       "impact": "serious",
>       "none": [],
>       "target": [
>         "ul"
>       ]
>     }
>   ],
>   "tags": [
>     "wcag2a",
>     "wcag412"
>   ]
> }
> 

## Environment

* Component Name and Version:
* Browser Name and Version:
* Node/npm Version: [e.g. Node 8/npm 5]
* Webpack Version:
* Operating System and version (desktop or mobile):
